### PR TITLE
2817 enable shorthands for lodash

### LINKED ISSUE #2817: web: "iteratee is not a function" error when using Node 12
## Bug Report
Building the dashboard with Node 12 results in a JavaScript error from the BaseTable component.

This is related to #2809 

### What is the issue?
BaseTable.jsx uses `lodash/sortBy` and using Node 12 appears to pull in a different version or cause a stricter check on function parameters.

### How can it be reproduced?
These instructions assume that the developer has a kubernetes cluster and linkerd is installed there.

Start with a clean environment.
```
rm -rf web/app/node_modules
```

Run the web setup
```
bin/web setup
```

Run the dev server (also works with the `run` command
```
bin/web dev
```

Open the dashboard in the browser:
http://localhost:7777

### Logs, error output, etc
The complete console output is [here](https://gist.github.com/cpretzer/ab2e716c4b41556df78cfc31814da857)

#### `linkerd check` output

kubernetes-api
--------------
√ can initialize the client
√ can query the Kubernetes API

kubernetes-version
------------------
√ is running the minimum Kubernetes API version
√ is running the minimum kubectl version

linkerd-existence
-----------------
√ control plane namespace exists
√ controller pod is running
√ can initialize the client
√ can query the control plane API

linkerd-api
-----------
√ control plane pods are ready
√ control plane self-check
√ [kubernetes] control plane can talk to Kubernetes
√ [prometheus] control plane can talk to Prometheus
√ no invalid service profiles

linkerd-version
---------------
√ can determine the latest version
√ cli is up-to-date

control-plane-version
---------------------
√ control plane is up-to-date
√ control plane and cli versions match

Status check results are √

### Environment

- Kubernetes Version:
v1.10.11

- Cluster Environment: (GKE, AKS, kops, ...)
Kubernetes on Docker for Mac

- Host OS:
macOS Mojave

- Linkerd version:
stable-2.3.0

### Possible solution
Set the `state.orderBy` at line 57 to an empty array if `this.props.defaultOrderBy` is undefined:

```
  constructor(props) {
    super(props);
    this.state = {
      order: this.props.defaultOrder || "asc",
      orderBy: this.props.defaultOrderBy || [],
      filterBy: ""
    };
    this.handleFilterInputChange = this.handleFilterInputChange.bind(this);
    this.handleFilterToggle = this.handleFilterToggle.bind(this);
  }
```

### PR DESCRIPTION
**Pull request by cpretzer**
Problem:
The parameter to sortBy is not properly interpreted as a function, and the `iteratee is not a function` error occurs.

Solution
Set `shorthand: true` in webpack.config.js

Validation
I tested with both node 10 and 12, including tests with `bin/web run` and `bin/web dev` for each version of node.

I have questions about the knock-on effects of setting shorthands to true, specifically around the size of the file that is generated. I ask that one or more of the reviewers validate that this is the right change.

It's possible that this fix as well as the fix for #2809 are not the optimal fixes.

Fixes #2817 

Notes:
This branch includes a merge from pretzer/2809-update-webpack-dev-server-version

### CODE REVIEW

**Comment by scottcarol** [comment]
Hey @cpretzer thanks so much for this PR! 👍 Can you give me instructions on replicating the `iteratee is not a function` error? Is this happening when you are attempting to sort on the dashboard or are you seeing it before then? Thanks!

**Comment by scottcarol** [comment]
Oops, I just saw that you had detailed output [here](https://github.com/linkerd/linkerd2/issues/2817). Let me try to replicate this on my end!

File: web/app/package.json [hunk 1]
```diff
@@ -68,7 +68,7 @@
     "webpack": "4.20.2",
     "webpack-bundle-analyzer": "^3.0.3",
     "webpack-cli": "3.0.7",
-    "webpack-dev-server": "3.1.14"
+    "webpack-dev-server": "^3.2.0"
```
Comment (rmars) [inline web/app/package.json]: can we continue with the version pinning already here? we can pin to `3.3.1` rather than `^`

**Comment by scottcarol** [comment]
Ah, thanks for the clarification on bundle size @rmars, I was comparing the actual size of `yarn.lock` in the master branch (401kb) vs this branch (421kb) in the terminal, but it makes sense to look at the whole bundle.

File: web/app/package.json (hunk 1 above)
Comment (cpretzer) [reply web/app/package.json]: No problem. I'll change to pin the version to 3.3.1.

**Comment by cpretzer** [comment]
@rmars The latest commit removes the lodash rules as well as the plugin

**Pull request merged**